lib/json_api/game_deserializer.rb

```diff
@@ -9,37 +10,52 @@ def initialize
       @player_deserializer = PlayerDeserializer.new
     end
 
-    def deserialize(serialized_game)
-      players = @player_deserializer.deserialize(serialized_game['players'])
-      current_player = deserialize_current_player(serialized_game['game'])
-      board_size = deserialize_board_size(serialized_game['game'])
-      board = deserialize_board(serialized_game['game'])
-      state = deserialize_state(serialized_game['game'])
-      Game::Game.new(players, current_player:current_player, board_size: board_size, board: board, state: state)
+    def deserialize(request_body)
+      raise_error_if_invalid(request_body)
+      players = @player_deserializer.deserialize(request_body)
+      Game::Game.new(players, deserialize_game(request_body['game']))
     end
 
     private
 
-    def deserialize_players(players)
-      [
-        deserialize_player(players['player_one']),
-        deserialize_player(players['player_two'])
-      ]
+    def raise_error_if_invalid(request_body)
+      raise DeserializationError.new("game is missing") if missing_game?(request_body)
+      raise DeserializationError.new("board is missing") if missing_board?(request_body['game'])
+      raise DeserializationError.new("current player is missing") if missing_current_player?(request_body['game'])
+      raise DeserializationError.new("state is missing") if missing_state?(request_body['game'])
     end
 
-    def deserialize_player(player)
-      name = player['name']
-      token = player['token']
-      type = player['type'].to_sym
-      Game::Player.create(type, token, name)
+    def missing_game?(request_body)
+      !request_body.has_key?('game')
+    end
+
+    def missing_board?(game)
+      !game.has_key?('board') || game['board'] == nil
```

Comment:
I think inverting this so it uses an 'and' would be cleaner so you're not doing the nil check on all of these

Reply:
@ssunday Would that not just equate to `!= nil`?

Reply:
It would but this feels harder to read. Like it's basically doing the invert on true on everything.

Reply:
Fair.